lib/json_schema/version_event_file_name.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require "initialize_with_config"
+
+module JsonSchema
+  class VersionEventFileName
+    include InitializeWithConfig
+    required_config :version, :event
+
+    def call
+      Rails.root.join(schema_root, version.to_s, schema_path, event.to_s, schema_file)
+    end
+
+  private
+
+    def default_config
+      {
+        schema_root: "etc/schema",
+        schema_path: "ecf/participant_declarations",
+        schema_file: "request_schema.json",
+      }
+    end
+  end
+end
```

Comment:
Soo, where do `version` and `event` live? Where do they need to live?

And what does `call` do exactly?

Reply:
Have a look at InitializeWithConfig. It's doing dependency injection (hence the required_config), via the configuration object passed from the calling services. Call in this instance is just creating a path to the schema file.

Reply:
Right, so they live on `VersionEventFileName`, so long as it's initialised with `version` and `event` arguments.

Reply:
Correct. The class level call with the appropriate parameters set as a configuration hash creates version and event methods which the instance level call uses.